docs/development/doc_guide.rst

```diff
@@ -1,76 +1,633 @@
-  use.
+       .. math::
 
-Substitutions
-=============
+          f(a, b) = a - b
+
+      Parameters
+      ----------
+      a : `float`
+          The left multiplicand.
+
+      b : `float`
+          The right multiplicand.
+
+      switch_order : `bool`, optional, keyword-only
+          If `True`, return :math:`a - b`. If `False`, then return
+          :math:`b - a`. Defaults to `True`.
+
+      Returns
+      -------
+      difference : float
+          The difference between ``a`` and ``b``.
+
+      Raises
+      ------
+      `ValueError`
+          If ``a`` or ``b`` is `~numpy.inf`.
+
+      Warns
+      -----
+      `UserWarning`
+          If ``a`` or ``b`` is `~numpy.nan`.
+
+      See Also
+      --------
+      add : Add two numbers.
+
+      Notes
+      -----
+      The "Notes" section provides extra information that cannot fit in
+      the extended summary near the beginning of the docstring. This
+      section should include a discussion of the physics behind a
+      particular concept that should be understandable to someone who is
+      taking their first plasma physics class. This section can
+      include a derivation of the quantity being calculated or a
+      description of a particular algorithm.
+
+      The next section contains example references to a journal article
+      [1]_ and a book [2]_.
+
+      References
+      ----------
+      .. [1] J. E. Foster, `Plasma-based water purification: Challenges and
+         prospects for the future <https://doi.org/10.1063/1.4977921>`_,
+         Physics of Plasmas, 22, 05501 (2017).
+
+      .. [2] E. Gamma, R. Helm, R. Johnson, J. Vlissides, `Design Patterns:
+         Elements of Reusable Object-Oriented Software
+         <https://www.oreilly.com/library/view/design-patterns-elements/0201633612/>`_
+
+      Examples
+      --------
+      Include a few example usages of the function here. Start with simple
+      examples and then increase complexity when necessary.
+
+      >>> from package.subpackage.module import subtract
+      >>> subtract(9, 6)
+      3
+
+      Here is an example of a multi-line function call.
+
+      >>> subtract(
+      ...     9, 6, switch_order=True,
+      ... )
+      -3
+
+      PlasmaPy's test suite will check that these commands provide the
+      output that follows each function call.
+      """
+      if np.isinf(a) or np.isinf(b):
+          raise ValueError("Cannot perform substraction operations involving infinity.")
+
+      warnings.warn("The subtract function encountered a nan value.", UserWarning)
+
+      return b - a if switch_order else a - b
+
+Template docstring
+~~~~~~~~~~~~~~~~~~
+
+This template docstring may be copied into new functions. Usually only
+some of the sections will be necessary for a particular function, but
+any sections that are included should be in the order provided.
+
+.. code-block:: python
+  :caption: Docstring template
+
+  def sample_function():
+      r"""
+      Return ...
```

Comment:
```suggestion
      Readable description of sample_function.
```

I don't want to encourage people to start of a docstring with "Return."  It redundant, just say what the functionality is doing.

Reply:
Huh.  I got that wording from one of the recommendations in [PEP 257](https://www.python.org/dev/peps/pep-0257/#id16).

Reply:
Yeah, I know.  It's widely used and in the PEP so I've never mentioned it.  You started removing them in a recent documentation refactor, so I'm hopeful that it gets propagated throughout the documentation.

Technically there's nothing wrong with it, but it's so redundant.  It doesn't do anything to expand the meaning of the documentation.  I already know a function is returning something, why are you telling me again!?  I have to read two extra words before I even get to the meat of the description.  It's just a pet-peeve of mine.  I won't use it but I also won't go throughout the documentation to strike it down.